command/agent/check.go

```diff
@@ -77,6 +81,10 @@ func (c *CheckType) IsTCP() bool {
 	return c.TCP != "" && c.Interval != 0
 }
 
+func (c *CheckType) IsDocker() bool {
+	return c.DockerContainerId != "" && c.Shell != "" && c.Interval != 0
```

Comment:
You probably don't want the shell check here since it's treated as optional down below.